rust/src/envs/linear_function.rs

```diff
@@ -204,6 +333,46 @@ impl Env for LinearFunction {
         .filter_map(|(index, &value)| if value { Some(index) } else { None }) // Collect indices where the value is true
         .collect()    
     }
+
+    fn twists(&self) -> (Vec<Vec<usize>>, Vec<Vec<usize>>) {
+        (self.obs_perms.clone(), self.act_perms.clone())
+    }
+}
+
+// metrics implementation shared via crate::envs::metrics
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn cx_gate_is_self_inverse() {
+        let gateset = vec![Gate::CX(0, 1)];
+        let metrics_weights = MetricsWeights::default();
+        let mut env = LinearFunction::new(2, 1, gateset, 2, 8, metrics_weights, true, true);
+        env.depth = env.max_depth;
+
+        env.step(0);
+        assert!(!env.solved());
```

Comment:
With `add_inverts=true`, each step() has 50% chance of inverting the state. The assertions can randomly fail, right?